Ruby-Assignment-2-LoopingsAndLiterals/Question-1/solution.rb

```diff
@@ -0,0 +1,44 @@
+
+INPUT_FILE_NAME = "./input.txt"
+
+# method for calculating sum of array elements
+def sum_of_numbers(int_array)
+  total = int_array.reduce(:+)
+  total ? total : 0
+end
+
+# method to check the array for the required condition
+def array_has_such_element(int_array)
+  int_array.each_with_index do |item, index|
+    left_subarray = int_array[0, index]
+    right_subarray = int_array[index+1, int_array.length]
+
+    if sum_of_numbers(left_subarray) == sum_of_numbers(right_subarray)
+      return true	
+    end
+  end
+  false
+end
+
+# read file
+input_file = File.open(INPUT_FILE_NAME)
+
+# read number of test cases
+TOTAL_CASES = input_file.readline.chomp.to_i
```

Comment:
Actually I was aware of it, but I thought of making input file name variable to be constant. and similarly total cases variable, since they won't change during the course of the program.